clis/teliod/src/command_listener.rs

```diff
@@ -0,0 +1,200 @@
+use crate::{
+    comms::DaemonConnection, daemon::TelioTaskCmd, ClientCmd, DaemonSocket, TelioStatusReport,
+    TeliodError,
+};
+use serde::{Deserialize, Serialize};
+use tokio::sync::mpsc;
+use tracing::{error, info};
+
+/// Command response type used to communicate between `telio runner -> daemon -> client`
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub enum CommandResponse {
+    Ok,
+    StatusReport(TelioStatusReport),
+    Err(String),
+}
+
+impl CommandResponse {
+    pub fn serialize(&self) -> String {
+        serde_json::to_string(self).unwrap_or("Err serializing".to_string())
+    }
+
+    pub fn deserialize(input: &str) -> Result<CommandResponse, TeliodError> {
+        serde_json::from_str::<CommandResponse>(input)
+            .map_err(|e| TeliodError::InvalidResponse(format!("{}", e)))
+    }
+}
+
+pub struct CommandListener {
+    socket: DaemonSocket,
+    /// Channel to send commands to telio task
+    telio_task_tx: mpsc::Sender<TelioTaskCmd>,
```

Comment:
nit: `mpsc::Sender/Receiver` is defined in `telio-task/src/io.rs` (`Rx` and `Tx`) so maybe it is worth using them.

Reply:
Good catch, I'll take a look

Reply:
Updated to use `chan::{Rx, Tx}`